src/store/modules/order/actions.ts

```diff
@@ -249,6 +249,75 @@ const actions: ActionTree<OrderState , RootState> ={
     return resp;
   },
 
+  async getCompletedOrders ({ commit, state }, payload) {
+    // Show loader only when new query and not the infinite scroll
+    if (payload.viewIndex === 0) emitter.emit("presentLoader");
+    let resp;
+    const orderQueryPayload = prepareOrderQuery({
+      ...payload,
+      shipmentMethodTypeId: !store.state.user.preference.showShippingOrders ? 'STOREPICKUP' : '',
+      orderItemStatusId: "ITEM_COMPLETED",
+      orderTypeId: 'SALES_ORDER',
+      docType: 'ORDER'
+    })
+
+    try {
+      resp = await OrderService.getCompletedOrders(orderQueryPayload)
+      if (resp.status === 200 && resp.data.grouped?.orderId?.ngroups > 0 && !hasError(resp)) {
+        let orders = resp?.data?.grouped?.orderId?.groups.map((order: any) => {
+          const orderItem = order.doclist.docs[0]
+          return {
+            orderId: orderItem.orderId,
+            orderName: orderItem.orderName,
+            customer: {
+              partyId: orderItem.customerPartyId,
+              name: orderItem.customerPartyName,
+            },
+            statusId: orderItem.orderStatusId,
+            parts: order.doclist.docs.reduce((arr: Array<any>, item: any) => {
+              const currentOrderPart = arr.find((orderPart: any) => orderPart.orderId === item.orderId)
```

Comment:
We should gather on the basis of parts only, but while rendering create a single item list